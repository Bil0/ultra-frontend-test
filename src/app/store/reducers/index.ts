import { MetaReducer } from '@ngrx/store';
import { environment } from 'environments/environment';

import * as fromSession from './marketplace.reducer';

export interface AppState {
  readonly marketplace: fromSession.MarketplaceState;
}

export const metaReducers: MetaReducer<AppState>[] = !environment.production
  ? []
  : [];
