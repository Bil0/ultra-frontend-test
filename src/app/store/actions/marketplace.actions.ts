import { createActionGroup, emptyProps, props } from '@ngrx/store';
import { HomeProduct } from 'app/models/product.model';
import { Wallet } from 'app/models/wallet.model';
import { Update } from '@ngrx/entity';

export const MarketplaceActions = createActionGroup({
  source: 'Marketplace',
  events: {
    'Load Wallet': emptyProps(),
    'Load Wallet Succeeded': props<{ wallet: Wallet }>(),
    'Load Wallet Error': props<{ error: unknown }>(),
    'Load Products': emptyProps(),
    'Load Products Succeeeded': props<{ products: HomeProduct[] }>(),
    'Load Product Error': props<{ error: unknown }>(),
    'Add To Basket': props<{ productId: number }>(),
    'Remove from Basket': props<{ productId: number }>(),
    'Reset State': emptyProps(),
  },
});
