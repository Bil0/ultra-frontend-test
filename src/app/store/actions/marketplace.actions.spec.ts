import { Action } from '@ngrx/store';

import { MarketplaceActions } from './marketplace.actions';

describe('Marketplace Actions', () => {
  let action: Action;

  test('Should create load wallet action', () => {
    action = MarketplaceActions.loadWallet();
    expect(action).toMatchSnapshot();
  });
});
