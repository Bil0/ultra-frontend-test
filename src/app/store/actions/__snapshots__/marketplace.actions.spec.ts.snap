// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marketplace Actions Should create load wallet action 1`] = `
Object {
  "type": "[Marketplace] Load Wallet",
}
`;
